### ISSUE TITLE
EmptyLineSeparatorCheck does not report violations when more than one empty line appears inside enum.

### ISSUE THREAD

**Issue by Praveen7294** [checkstyle/checkstyle#18437]
I have read check documentation: https://checkstyle.org/checks/whitespace/emptylineseparator.html#EmptyLineSeparator
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

#### Config:
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="EmptyLineSeparator">
      <property name="tokens"
               value="PACKAGE_DEF, IMPORT, STATIC_IMPORT, CLASS_DEF, INTERFACE_DEF, ENUM_DEF,
                    STATIC_INIT, INSTANCE_INIT, METHOD_DEF, CTOR_DEF, VARIABLE_DEF, RECORD_DEF,
                    COMPACT_CTOR_DEF"/>
      <property name="allowNoEmptyLineBetweenFields" value="true"/>
      <property name="allowMultipleEmptyLines" value="false"/>
      <property name="allowMultipleEmptyLinesInsideClassMembers" value="false"/>
    </module>
  </module>
</module>
```

#### Test.java:
```
public class Test {
  
  enum Status {


    STARTED,


    RUNNING,


    FINISHED


  }
}
```

#### Actual Cli:
```
$ java -jar checkstyle-12.3.0-all.jar -c config_check.xml Test.java
Starting audit...
Audit done.
```

#### Expected Cli:
```
$ java -jar checkstyle-12.3.0-all.jar -c config_check.xml Test.java
Starting audit...
[ERROR] F:\GitHub\headhtmltagname\Test.java:3:3: There is more than 1 empty line after this line. [EmptyLineSeparator]
[ERROR] F:\GitHub\headhtmltagname\Test.java:6:5: There is more than 1 empty line after this line. [EmptyLineSeparator]
[ERROR] F:\GitHub\headhtmltagname\Test.java:9:5: There is more than 1 empty line after this line. [EmptyLineSeparator]
[ERROR] F:\GitHub\headhtmltagname\Test.java:12:5: There is more than 1 empty line after this line. [EmptyLineSeparator]
Audit done.
```

Since the documentation mentions that `the check is valid only for statements that have a body—CLASS_DEF, INTERFACE_DEF, ENUM_DEF, STATIC_INIT, INSTANCE_INIT, METHOD_DEF, and CTOR_DEF`, it must also check inside enums.
https://checkstyle.org/checks/whitespace/emptylineseparator.html#Example5-config

**Comment by Praveen7294** [comment]
google formatter also support this
```
$ diff -Naru Test.java FormattedCode.java
--- Test.java   2026-01-07 00:49:22.224873600 +0530
+++ FormattedCode.java  2026-01-07 00:49:41.464285900 +0530
@@ -1,17 +1,13 @@
 public class Test {

   public enum Check {
-
     FIRST,
     SECOND,

     THIRD,

-
     FOURTH,

-
-
     SPECIAL(100) {
       @Override
       public String describe() {
@@ -19,17 +15,11 @@
       }
     },

-
-
-
     NORMAL(200) {
       @Override
       public String describe() {
-          return "Normal enum constant";
-        }
-      };
-
-
-
+        return "Normal enum constant";
+      }
+    };
   }
 }
```

**Comment by romani** [comment]
it might be a reason that ENUM_CONSTANT_DEF is not list of lowed tokens, so enum content is simply skipped.
but should validate extra lines before enum methods.
@Praveen7294 , please try.


should we consider this issue as google style defect ?

**Comment by Praveen7294** [comment]
> it might be a reason that ENUM_CONSTANT_DEF is not list of lowed tokens, so enum content is simply skipped.

You are right. `ENUM_CONSTANT_DEF` is not in the list of `tokens` in the configuration, and this token is also not included in the `tokens` property of the `EmptyLineSeparator` check.

> but should validate extra lines before enum methods.

I checked it. It validates extra lines before and after enum methods, constructors, and fields, and it even validates extra lines inside enum members. But it does not validate extra lines before and after enum constants.

we should add `ENUM_CONSTANT_DEF` token to validate extra line before and after enum constant.

In the Google Java Style Guide, it is mentioned that `After the comma that follows an enum constant, a line break is optional. Additional blank lines (usually just one) are also allowed.`
https://google.github.io/styleguide/javaguide.html#s4.8.1-enum-classes

> should we consider this issue as google style defect ?

First, we should add the `ENUM_CONSTANT_DEF` token to the `EmptyLineSeparator` check. Then, we will add this token to the `google_checks.xml` file.

**Comment by Praveen7294** [comment]
@romani I have updated the title, example, and description of the issue after validating blank lines in enums to make the issue more accurate.

**Comment by Praveen7294** [comment]
@romani After https://github.com/checkstyle/checkstyle/issues/18265#issuecomment-3719342677, please remove google style label this change will not add in `google_checks.xml` but we have to add `ENUM_CONSTANT_DEF` token in `EmptyLineSeparator` check to solve false negative.

**Comment by romani** [comment]
>google formatter also support this

I removed label, this comment confusing and looks like targeting google style.
BUT I am ok to just add new add ENUM_CONSTANT_DEF token in EmptyLineSeparator.

and in separate issue add this token to google_check.xml
